src/BaselineUtils.cc

```diff
@@ -463,16 +550,23 @@ _find_stray_flux(det::Footprint const& foot,
                 }
                 // the stray flux to give to template i
                 double p = (contrib[i] / csum) * (*in_it).image();
+
                 if (!strayfoot[i]) {
-                    strayfoot[i] = PTR(det::Footprint)(new det::Footprint(foot.getPeaks().getSchema()));
+                    strayfoot[i] = std::make_shared<det::Footprint>();
+                    strayfoot[i]->setPeakSchema(foot.getPeaks().getSchema());
                 }
-                strayfoot[i]->addSpanInSeries(y, x, x); // XXX this may be rather heavy in memory use
+                straySpans[i].push_back(geom::Span(y, x, x));
                 straypix[i].push_back(p);
                 straymask[i].push_back((*in_it).mask());
                 strayvar[i].push_back((*in_it).variance());
             }
         }
     }
+    for (std::size_t i= 0; i < tfoots.size(); ++i) {
+        if (strayfoot[i]) {
+            strayfoot[i]->setSpans(std::make_shared<geom::SpanSet>(straySpans[i]));
+        }
+    }
```

Comment:
good catch. I was too focused on fixing the above loop, I didn't look at what was below